endpoints/openrtb2/amp_auction.go

```diff
@@ -443,7 +444,11 @@ func (deps *endpointDeps) overrideWithParams(ampParams amp.Params, req *openrtb2
 		req.Imp[0].TagID = ampParams.Slot
 	}
 
-	policyWriter, policyWriterErr := readPolicy(ampParams.Consent)
+	policy := policyReader{
```

Comment:
I think it's good to make the policy reader more open, but please define it earlier and pass it down as a dependency instead of creating it right here just once when we could reuse for every auction.

Reply:
Good catch. Corrected